grafana_client/elements/datasource.py

```diff
@@ -452,10 +490,16 @@ def health_check(self, datasource: Union[DatasourceIdentifier, Dict]) -> Datasou
                     message = f"Invalid response. {reason}"
 
             elif datasource_type == "loki":
-                if "status" in response and response["status"] == "success":
+                if self.api.version and VERSION_7 <= LooseVersion(self.api.version) < VERSION_8:
+                    if "status" in response and response["status"] == "success":
+                        message = "Success"
+                        success = True
+                    else:
+                        message = response.get("message", "Unknown error")
+                elif "results" in response and "test" in response["results"]:
                     message = "Success"
                     success = True
-                else:
+                elif "message" in response:
                     message = response["message"]
```

Comment:
I've adjusted this a bit, compared to your version, @peekjef72. When anything is wrong, please send fixes. 🌻